pkg/engines/tekton/controller.go

```diff
@@ -1,743 +1,136 @@
 	}
-	return nil
-}
-
-func trimDashboardURL(base string) string {
-	return strings.TrimSuffix(strings.TrimSuffix(base, "#"), "/")
-}
-
-// RateLimiter creates a ratelimiting queue for the foghorn controller.
-func RateLimiter() workqueue.RateLimitingInterface {
-	rl := workqueue.NewMaxOfRateLimiter(
-		workqueue.NewItemExponentialFailureRateLimiter(5*time.Millisecond, 120*time.Second),
-		&workqueue.BucketRateLimiter{Limiter: rate.NewLimiter(rate.Limit(1000), 50000)},
-	)
-	return workqueue.NewNamedRateLimitingQueue(rl, controllerName)
-}
 
-func createLabelSelectorFromActivity(activity *v1alpha1.ActivityRecord) (labels.Selector, error) {
-	var selectors []string
-
-	if activity.Owner != "" {
-		selectors = append(selectors, fmt.Sprintf("%s=%s", util.OrgLabel, strings.ToLower(activity.Owner)))
-	}
-	if activity.Repo != "" {
-		selectors = append(selectors, fmt.Sprintf("%s=%s", util.RepoLabel, activity.Repo))
-	}
-	if activity.Branch != "" {
-		selectors = append(selectors, fmt.Sprintf("%s=%s", util.BranchLabel, activity.Branch))
-	}
-	if activity.BuildIdentifier != "" {
-		selectors = append(selectors, fmt.Sprintf("%s=%s", util.BuildNumLabel, activity.BuildIdentifier))
-	}
-	if activity.Context != "" {
-		selectors = append(selectors, fmt.Sprintf("%s=%s", util.ContextLabel, activity.Context))
+	// get job's pipeline runs
+	var pipelineRunList pipelinev1beta1.PipelineRunList
+	if err := r.client.List(ctx, &pipelineRunList, client.InNamespace(req.Namespace), client.MatchingFields{jobOwnerKey: req.Name}); err != nil {
+		r.logger.Errorf("Failed list pipeline runs: %s", err)
+		return ctrl.Result{}, err
 	}
 
-	return labels.Parse(strings.Join(selectors, ","))
-}
-
-// generateBuildID generates a unique build ID for consistency with Prow behavior
-func generateBuildID() string {
-	return fmt.Sprintf("%d", utilrand.Int())
-}
-
-// makePipeline creates a PipelineRun and substitutes LighthouseJob managed pipeline resources with ResourceSpec instead of ResourceRef
-// so that we don't have to take care of potentially dangling created pipeline resources.
-func (c *Controller) makePipelineRun(lj v1alpha1.LighthouseJob) (*tektonv1beta1.PipelineRun, error) {
-	// First validate.
-	if lj.Spec.PipelineRunSpec == nil {
-		return nil, errors.New("no PipelineSpec defined")
-	}
-
-	buildID := generateBuildID()
-	if buildID == "" {
-		return nil, errors.New("empty BuildID in status")
-	}
-
-	prLabels, annotations := jobutil.LabelsAndAnnotationsForJob(lj, buildID)
-	specCopy := lj.Spec.PipelineRunSpec.DeepCopy()
-	p := tektonv1beta1.PipelineRun{
-		ObjectMeta: metav1.ObjectMeta{
-			Annotations:  annotations,
-			GenerateName: fmt.Sprintf("%s-", lj.Spec.Job),
-			Namespace:    lj.Spec.Namespace,
-			Labels:       prLabels,
-		},
-		Spec: *specCopy,
-	}
-	// Set a default timeout of 1 day if no timeout is specified
-	if p.Spec.Timeout == nil {
-		p.Spec.Timeout = &metav1.Duration{Duration: 24 * time.Hour}
-	}
-	p.OwnerReferences = []metav1.OwnerReference{{
```

Comment:
I need to take care here as the `Controller` property was not set. Therefore it won't be considered by the indexer.

Reply:
> I'm fine with things not quite working right for jobs that are running at upgrade time - there aren't enough people using Lighthouse + Tekton in production at this point for it to be a major problem, so let's go ahead and do it right going forward without excessive complexity for backwards compat.

👍